src/attributes.jl

```diff
@@ -850,6 +850,18 @@ struct TimeLimitSec <: AbstractOptimizerAttribute end
 
 attribute_value_type(::TimeLimitSec) = Union{Nothing,Float64}
 
+"""
+    ObjectiveLimit()
+
+
+An optimizer attribute for setting a limit on the objective value.
+The solver may stop when the `ObjectiveValue` is better (lower for minimization, higher for maximization) than the `ObjectiveLimit`.
+When `set` to `nothing`, it removes the solver limit. The default value is `nothing`.
+"""
+struct ObjectiveLimit <: AbstractOptimizerAttribute end
+
+attribute_value_type(::ObjectiveLimit) = Union{Nothing,Float64}
```

Comment:
An implicit assumption is that this applies only to scalar-valued objective functions, and that the objective is `Float64`. What if the optimizer is `BigFloat`, or `Rational{BigFloat}`? And what about vector-valued objectives?

Reply:
vetor-valued objectives is another story I wouldn't touch at first. I would be okay upgrading this to:
```
Union{Nothing,Real}
```

Reply:
Maybe we should just remove this for now. It's only implemented for a subset of attributes, and the fallback is `Any`.

Reply:
Sure yes